python/semantic_kernel/connectors/memory/azure_cosmosdb_no_sql/azure_cosmosdb_no_sql_memory_store.py

```diff
@@ -0,0 +1,184 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import json
+from typing import Any, Dict, List, Tuple
+
+import numpy as np
+from azure.cosmos.aio import ContainerProxy, CosmosClient, DatabaseProxy
+from numpy import ndarray
+from python.semantic_kernel.exceptions.service_exceptions import ServiceInitializationError
+
+from semantic_kernel.memory.memory_record import MemoryRecord
+from semantic_kernel.memory.memory_store_base import MemoryStoreBase
+
+
+class AzureCosmosDBNoSQLMemoryStore(MemoryStoreBase):
+    cosmos_client: CosmosClient = None
+    database: DatabaseProxy
+    container: ContainerProxy
+    database_name: str = None
+    partition_key: str = None
+    vector_embedding_policy: [Dict[str, Any]] = None
+    indexing_policy: [Dict[str, Any]] = None
+    cosmos_container_properties: [Dict[str, Any]] = None
+
+    def __init__(
+        self,
+        cosmos_client: CosmosClient,
+        database_name: str,
+        partition_key: str,
+        vector_embedding_policy: [Dict[str, Any]],
+        indexing_policy: [Dict[str, Any]],
+        cosmos_container_properties: [Dict[str, Any]],
+    ):
+        if indexing_policy["vectorIndexes"] is None or len(indexing_policy["vectorIndexes"]):
+            raise ServiceInitializationError("Vector dimensions must be a positive number.")
+
+        self.cosmos_client = cosmos_client
+        self.database_name = database_name
+        self.partition_key = partition_key
+        self.vector_embedding_policy = vector_embedding_policy
+        self.indexing_policy = indexing_policy
+        self.cosmos_container_properties = cosmos_container_properties
```

Comment:
Note: if you pull in main, we just implemented Pydantic settings, which would allow one to read these values from env vars/.env file. Just a note for you.

Reply:
Those are dicts, not just values which can be read from env vars/.env file.